test/src/test/java/org/corfudb/integration/CorfuReplicationMultiSenderIT.java

```diff
@@ -0,0 +1,588 @@
+import org.corfudb.runtime.collections.CorfuStreamEntries;
+import org.corfudb.runtime.collections.CorfuStreamEntry;
+import org.corfudb.runtime.collections.StreamListener;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.test.SampleSchema;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.corfudb.infrastructure.logreplication.replication.receive.LogReplicationMetadataManager.LR_STATUS_STREAM_TAG;
+
+@Slf4j
+@RunWith(Parameterized.class)
+public class CorfuReplicationMultiSenderIT extends AbstractIT {
+    private final int sourceSiteCorfuPort1 = 9000;
+    private final int sourceSiteCorfuPort2 = 9002;
+    private final int sourceSiteCorfuPort3 = 9004;
+    private final int sinkSiteCorfuPort = 9001;
+    private final int sourceReplicationPort1 = 9010;
+    private final int sourceReplicationPort2 = 9011;
+    private final int sourceReplicationPort3 = 9012;
+
+    private final int sinkReplicationPort = 9020;
+    private Process sourceCorfu1 = null;
+    private Process sourceCorfu2 = null;
+    private Process sourceCorfu3 = null;
+    private Process sinkCorfu = null;
+    private Process sourceReplicationServer1 = null;
+    private Process sourceReplicationServer2 = null;
+    private Process sourceReplicationServer3 = null;
+    private Process sinkReplicationServer = null;
+
+    private CorfuRuntime sourceRuntime1;
+    private CorfuRuntime sourceRuntime2;
+    private CorfuRuntime sourceRuntime3;
+    private CorfuRuntime sinkRuntime;
+
+    private CorfuStore corfuStoreSource1;
+    private CorfuStore corfuStoreSource2;
+    private CorfuStore corfuStoreSource3;
+    private CorfuStore corfuStoreSink;
+
+    private final String sourceEndpoint1 = DEFAULT_HOST + ":" + sourceSiteCorfuPort1;
+
+    private final String sourceEndpoint2 = DEFAULT_HOST + ":" + sourceSiteCorfuPort2;
+    private final String sourceEndpoint3 = DEFAULT_HOST + ":" + sourceSiteCorfuPort3;
+    private final String sinkEndpoint = DEFAULT_HOST + ":" + sinkSiteCorfuPort;
+
+    private static final String TABLE_1 = "Table001";
+    private static final String TABLE_2 = "Table002";
+    private static final String TABLE_3 = "Table003";
+
+    private static final String TABLE_MERGED = "Table_Merged";
+    private static final String NAMESPACE = DefaultLogReplicationConfigAdapter.NAMESPACE;
+
+    private static final String STREAM_TAG = DefaultLogReplicationConfigAdapter.TAG_ONE;
+
+    private static final int THREE = 3;
```

Comment:
Same as comment in previous file, it is better to have meaningful variables, gives better meaning to the test and eases readability.